packages/CalendarCarousel/index.tsx

```diff
@@ -1,19 +1,274 @@
-import { StyleSheet, Text, View } from 'react-native';
+import {
+  FlatList,
+  SafeAreaView,
+  ScrollView,
+  StyleSheet,
+  Text,
+  TextStyle,
+  TouchableOpacity,
+  View,
+  ViewStyle,
+} from 'react-native';
+import React, { ReactElement } from 'react';
 
-import React from 'react';
+interface Style {
+  container: ViewStyle;
+  headerFlex: ViewStyle;
+  beforeArrowText: TextStyle;
+  titleFlex: ViewStyle;
+  titleText: TextStyle;
+  yearText: TextStyle;
+  nextArrowText: TextStyle;
+  weekdayFlex: ViewStyle;
+  weekdayText: TextStyle;
+  calendarContainerSize: ViewStyle;
+}
 
-const styles = StyleSheet.create({
+const styles = StyleSheet.create<Style>({
   container: {
     flex: 1,
+    paddingTop: 40,
+    width: 330,
+    height: 388,
+  },
+  headerFlex: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    paddingBottom: 30,
+  },
+  beforeArrowText: {
+    color: 'royalblue',
+    fontSize: 30,
+  },
+  titleFlex: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  titleText: {
+    fontFamily: 'Avenir',
+    fontSize: 20,
+    textAlign: 'center',
+    justifyContent: 'center',
+    width: 300,
+  },
+  yearText: {
+    fontSize: 12,
+    textAlign: 'center',
+    justifyContent: 'center',
+  },
+  nextArrowText: {
+    color: 'royalblue',
+    fontSize: 30,
+  },
+  weekdayFlex: {
+    flexDirection: 'row',
+  },
+  weekdayText: {
+    color: '#4F4F4F',
+    fontFamily: 'Avenir',
+    fontSize: 20,
+    textAlign: 'center',
+  },
+  calendarContainerSize: {
+    width: 330,
+    height: 388,
   },
 });
 
-function Item(): React.ReactElement {
+const data = {
+  dates: [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31],
+  week: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
```

Comment:
This also should be done programmatically to cover cases like Arabic calendar which starts with `monday`.

![image](https://user-images.githubusercontent.com/27461460/87587237-cec12e80-c71c-11ea-8cf6-f55426c26cb1.png)